src/X.PagedList/BasePagedList.cs

```diff
@@ -103,25 +102,109 @@ IEnumerator IEnumerable.GetEnumerator()
+    /// 	Maximum size any individual subset.
+    /// </summary>
+    /// <value>
+    /// 	Maximum size any individual subset.
+    /// </value>
+    public int PageSize { get; protected set; }
+
+    /// <summary>
+    /// 	Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and this is NOT the first subset within the superset.
+    /// </summary>
+    /// <value>
+    /// 	Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and this is NOT the first subset within the superset.
+    /// </value>
+    public bool HasPreviousPage { get; protected set; }
+
+    /// <summary>
+    /// Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and this
+    /// is NOT the last subset within the superset.
+    /// </summary>
+    /// <value>
+    /// Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and this
+    /// is NOT the last subset within the superset.
+    /// </value>
+    public bool HasNextPage { get; protected set; }
+
+    /// <summary>
+    /// Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and this
+    /// is the first subset within the superset.
+    /// </summary>
+    /// <value>
+    /// Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and
+    /// this is the first subset within the superset.
+    /// </value>
+    public bool IsFirstPage { get; protected set; }
+
+    /// <summary>
+    /// Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and
+    /// this is the last subset within the superset.
+    /// </summary>
+    /// <value>
+    /// Returns true if the superset is not empty and PageNumber is less than or equal to PageCount and this
+    /// is the last subset within the superset.
+    /// </value>
+    public bool IsLastPage { get; protected set; }
+
+    /// <summary>
+    /// One-based index of the first item in the paged subset, zero if the superset is empty or PageNumber
+    /// is greater than PageCount.
+    /// </summary>
+    /// <value>
+    /// One-based index of the first item in the paged subset, zero if the superset is empty or PageNumber
+    /// is greater than PageCount.
+    /// </value>
+    public int FirstItemOnPage { get; protected set; }
+
+    /// <summary>
+    /// One-based index of the last item in the paged subset, zero if the superset is empty or PageNumber
+    /// is greater than PageCount.
+    /// </summary>
+    /// <value>
+    /// One-based index of the last item in the paged subset, zero if the superset is empty or PageNumber
+    /// is greater than PageCount.
+    /// </value>
+    public int LastItemOnPage { get; protected set; }
+
+    public IPagedList GetMetaData()
     {
-        return new PagedListMetaData(this);
+        return new StaticPagedList<object>(ImmutableArray<object>.Empty, this);
```

Comment:
We also need to adjust the summary for StaticPagedList(IEnumerable<T>, IPagedList), because it tells stuff about a "MetaData" property.

Personally, I would consider to drop that particular constructor of StaticPagedList entirely, though, and use the other one instead. I do not like to hand over an object and then only use a small fraction of its properties.

Furthermore, this is a behavior change. PagedListMetaData copied all of the properties, StaticPagedList with the given constructor only keeps three of them. Anyway, if we only return an IPagedList from an IPagedList here, why do we copy at all? I would drop this GetMetaData() method entirely instead. If somebody needs to properties, he can use the original reference as source. If somebody needs a copy, we either need to provide a proper copy constructor for all the relevant classes or we make it the user's problem. I would go with the latter, since the user will know better which particular subset of variables he will need beyond the IPagedList type itself.

Reply:
Agree. I also think that GetMetaData can be removed in future versions.